src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Capture.cs

```diff
@@ -7,15 +7,24 @@
 // search may return multiple Capture within each capturing
 // RegexGroup.
 
+using System.Buffers;
+
 namespace System.Text.RegularExpressions
 {
     /// <summary>
     /// Represents the results from a single subexpression capture. The object represents
-    /// one substring for a single successful capture.
+    /// the input slice for a single successful capture.
     /// </summary>
     public class Capture
     {
-        internal Capture(string text, int index, int length)
+        private string _value;
+
+        /// <summary>
+        /// Creates a Capture with an input string, a start index of the capture
+        /// and the length of the captured string.
+        /// </summary>
+        /// <param name="text">The input string provided by the user wrapped in a Memory.</param>
+        internal Capture(in MemoryOrPinnedSpan<char> text, int index, int length)
         {
             Text = text;
             Index = index;
```

Comment:
Why are a separate index and length needed, rather than just slicing the MemoryOrPinnedSpan?

Reply:
Because the index of the captured string is needed by Regex.Replace and Regex.Split.